Restructure significant part of the controller documentation (#12549)
nextcloud/documentation: developer_manual/basics/routing.rst

* **postfix** (Optional): lets you define a route id postfix. Since each route name will be transformed to a route id (**page#method** -> **myapp.page.method**) and the route id can only exist once you can use the postfix option to alter the route id creation by adding a string to the route id, e.g., **'name' => 'page#method', 'postfix' => 'test'** will yield the route id **myapp.page.methodtest**. This makes it possible to add more than one route/URL for a controller method
* **defaults** (Optional): If this setting is given, a default value will be assumed for each URL parameter which is not present. The default values are passed in as a key => value pair array

.. _routes_ocs:

OCS routes
----------

Registering of OCS routes requires to use a :ref:`corresponding controller<ocscontroller>`.
Additionally, the route must be configured to be an OCS route in the router.
To do so, you use the ``ocs`` key in the ``routes.php`` file instead of the key ``routes``.
The rest of the structure is the same.

> You can of course combine non-OCS with OCS routes.

Reviewer: ```suggestion
You can of course have both index.php and OCS routes.
```
Combining doesn't really make sense in this context.
